docs/branching.md

```diff
@@ -0,0 +1,32 @@
+### Branching and deployments model
+
+This document aims to discuss and document the relations between branches and deployments of `elixir-omg`, with respect to branches and deployments of `plasma-contracts`.
+
+It is a refinement of the [OIP4 branching model](https://github.com/omisego/OIP/blob/master/0004-ewallet-release-and-versioning.md), applicable to `elixir-omg` and `plasma-contracts` versioning.
+
+This takes `v0.1` and `v0.2` as an example of two versions, where one is deployed in `staging`/`Ari` and the other is the upcoming upgrade and is deployed in `devel`.
+
+Rationale:
+- the history and the relations between the versions are readable and accountable,
+- we can predictably sync our respective watchers/run child chain servers against deployed contracts
+- and at the same time be able to move with `master`s quickly
+
+("->" means that the `mix.exs` holds the following branch)
+
+- `elixir-omg/v0.1` -> `plasma-contracts/v0.1`, that's understood, the `mix.lock` points to a specific frozen version, no changes expected there
+- `elixir-omg/master` -> `plasma-contracts/master`, mix lock points to the contracts ~deployed on `devel`~ we want to CI-check with `elixir-omg`. So it might be newer than what contract is deployed on `devel`
```

Comment:
Confusing language here.

> mix lock points to the contracts ~deployed on `devel`~

and

> it might be newer than what contract is deployed on `devel`

Are you trying to say that we will try to point to the contract deployed on the `development` environment, except contract deployment is a manual process right now.

Ideally, for `development` environment, the contract gets re-deployed when there is a dependency update. This requires some tooling work and probably want to decouple contract deployments out of the elixir-omg before doing this.

Reply:
Sorry, missed the markdown strike-through. Ignore the first part.

Reply:
Yes, that's what I mean, to have the contract re-deploy automatically. There's some yak shaving we need to do to decouple the deployment tooling.